101-mongodb-sharded-on-centos/azuredeploy.json

```diff
@@ -0,0 +1,582 @@
+          "base": "10.0.0.",
+          "start": 5
+        },
+        "configPrimaryIp": "10.0.0.240",
+        "configSecondaryIp1": "10.0.0.241",
+        "configSecondaryIp2": "10.0.0.242",
+        "routerIp": "10.0.0.230",
+        "replicaPrimaryIp": "10.0.0.100"
+      }
+    },
+    "configPrimaryOsSettings": {
+      "imageReference": {
+        "publisher": "OpenLogic",
+        "offer": "CentOS",
+        "sku": "[parameters('centOsVersion')]",
+        "version": "latest"
+      },
+      "scripts": [
+        "[variables('configPrimaryNodeScript')]"
+      ],
+      "certificates": "[variables('certificateLocation')]"
+    },
+    "configSecondaryOsSettings": {
+      "imageReference": {
+        "publisher": "OpenLogic",
+        "offer": "CentOS",
+        "sku": "[parameters('centOsVersion')]",
+        "version": "latest"
+      },
+      "scripts": [
+        "[variables('configSecondaryNodeScript')]"
+      ],
+      "certificates": "[variables('certificateLocation')]"
+    },
+    "routerOsSettings": {
+      "imageReference": {
+        "publisher": "OpenLogic",
+        "offer": "CentOS",
+        "sku": "[parameters('centOsVersion')]",
+        "version": "latest"
+      },
+      "scripts": [
+        "[variables('routerNodeScript')]"
+      ],
+      "certificates": "[variables('certificateLocation')]"
+    },
+    "replicaPrimaryOsSettings": {
+      "imageReference": {
+        "publisher": "OpenLogic",
+        "offer": "CentOS",
+        "sku": "[parameters('centOsVersion')]",
+        "version": "latest"
+      },
+      "scripts": [
+        "[variables('replicaPrimaryNodeScript')]"
+      ],
+      "certificates": "[variables('certificateLocation')]"
+    },
+    "replicaSecondaryOsSettings": {
+      "imageReference": {
+        "publisher": "OpenLogic",
+        "offer": "CentOS",
+        "sku": "[parameters('centOsVersion')]",
+        "version": "latest"
+      },
+      "scripts": [
+        "[variables('replicaSecondaryNodeScript')]"
+      ],
+      "certificates": "[variables('certificateLocation')]"
+    }
+  },
+  "resources": [
+    {
+      "name": "shared",
+      "type": "Microsoft.Resources/deployments",
+      "apiVersion": "2019-07-01",
+      "properties": {
+        "mode": "Incremental",
+        "templateLink": {
+          "uri": "[variables('sharedTemplateUrl')]",
+          "contentVersion": "1.0.0.0"
+        },
+        "parameters": {
+          "networkSettings": {
+            "value": "[variables('networkSettings')]"
+          },
+          "namespace": {
+            "value": "[variables('namespace')]"
+          },
+          "location": {
+            "value": "[parameters('location')]"
+          }
+        }
+      }
+    },
+    {
+      "name": "[concat('configSecondaryNode', copyIndex(1))]",
+      "type": "Microsoft.Resources/deployments",
+      "apiVersion": "2019-07-01",
+      "dependsOn": [
+        "[concat('Microsoft.Resources/deployments/', 'shared')]"
```

Comment:
use resourceId for the dependsOn throughout - or you can just use the deployment name as you have in some of the resources